Assets/HoloToolkit/Input/Scripts/Microphone/DictationInputManager.cs

```diff
@@ -0,0 +1,254 @@
+        /// </summary>
+        public static bool IsListening { get; private set; }
+
+        /// <summary>
+        /// String result of the current dictation.
+        /// </summary>
+        private static string dictationResult;
+
+        /// <summary>
+        /// Audio clip of the last dictation session.
+        /// </summary>
+        private static AudioClip dictationAudioClip;
+
+        private static DictationRecognizer dictationRecognizer;
+
+        private bool hasFailed;
+
+        #region Unity Methods
+
+        protected override void Awake()
+        {
+            base.Awake();
+
+            dictationResult = string.Empty;
+
+            dictationRecognizer = new DictationRecognizer();
+            dictationRecognizer.DictationHypothesis += DictationRecognizer_DictationHypothesis;
+            dictationRecognizer.DictationResult += DictationRecognizer_DictationResult;
+            dictationRecognizer.DictationComplete += DictationRecognizer_DictationComplete;
+            dictationRecognizer.DictationError += DictationRecognizer_DictationError;
+
+            // Query the maximum frequency of the default microphone.
+            int minSamplingRate; // Unsued.
+            Microphone.GetDeviceCaps(DeviceName, out minSamplingRate, out samplingRate);
+        }
+
+        private void LateUpdate()
+        {
+            if (IsListening && !Microphone.IsRecording(DeviceName) && dictationRecognizer.Status == SpeechSystemStatus.Running)
+            {
+                // If the microphone stops as a result of timing out, make sure to manually stop the dictation recognizer.
+                StartCoroutine(StopRecording());
+            }
+
+            if (!hasFailed && dictationRecognizer.Status == SpeechSystemStatus.Failed)
+            {
+                hasFailed = true;
+                InputManager.Instance.RaiseDictationError(Instance, 0, "Dictation recognizer has failed!");
+            }
+        }
+
+        protected override void OnDestroy()
+        {
+            dictationRecognizer.Dispose();
+
+            base.OnDestroy();
+        }
+
+        #endregion // Unity Methods
+
+        /// <summary>
+        /// Turns on the dictation recognizer and begins recording audio from the default microphone.
+        /// </summary>
+        /// <param name="initialSilenceTimeout">The time length in seconds before dictation recognizer session ends due to lack of audio input in case there was no audio heard in the current session.</param>
+        /// <param name="autoSilenceTimeout">The time length in seconds before dictation recognizer session ends due to lack of audio input.</param>
+        /// <param name="recordingTime">Length in seconds for the manager to listen.</param>
+        /// <returns></returns>
+        public static IEnumerator StartRecording(float initialSilenceTimeout = 5f, float autoSilenceTimeout = 20f, int recordingTime = 10)
+        {
+            if (IsListening) { yield break; }
```

Comment:
I'll go with option 1. Seems a bit easier to implement.  Although, I don't thing the button would need to track the return value.